File: Convert Sorted Array to Binary Search Tree.cpp
#include<iostream>

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};


TreeNode *BuildTree(vector<int> &A,int l,int r)
    {
        if(l>r) return NULL;
        int mid=(l+r)/2;
        TreeNode *node = new TreeNode(A[mid]);
        node->left=BuildTree(A, l, mid-1);
        node->right=BuildTree(A, mid+1, r);
        return node;
    }

TreeNode *sortedArrayToBST(vector<int> &num)
    {
        if(num.size()==0) return NULL;
        int len=num.size();
        return BuildTree(num,0,len-1);
    }

File: Python/Convert Sorted Array to Binary Search Tree.py
# Definition for a  binary tree node
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    # @param num, a list of integers
    # @return a tree node
    def MakeTree(self, num, left, right):
        if left > right:
            return None
        mid = (left + right)/2
        root = TreeNode(num[mid])
        root.left = self.MakeTree(num, left, mid-1);
        root.right = self.MakeTree(num, mid+1, right);
        return root
    def sortedArrayToBST(self, num):
        if len(num) == 0:
            return None
        root = self.MakeTree(num, 0, len(num)-1)
        return root